components/brave_savings/browser/perf_predictor_web_contents_observer.cc

```diff
@@ -0,0 +1,160 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_savings/browser/perf_predictor_web_contents_observer.h"
+
+#include "base/memory/singleton.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/sessions/session_tab_helper.h"
+#if !defined(OS_ANDROID)
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_list.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+#endif
+#include "components/prefs/pref_service.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "content/public/browser/navigation_entry.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_contents_user_data.h"
+#include "content/public/common/resource_type.h"
+#include "brave/common/pref_names.h"
+#include "brave/vendor/brave_performance_predictor/src/third_party_extractor.h"
+#include "brave/vendor/brave_performance_predictor/src/third_parties.h"
+
+using content::ResourceType;
+using base::Singleton;
+using content::WebContents;
+
+namespace brave_perf_predictor {
+
+class ThirdPartyExtractor;
+
+PerfPredictorWebContentsObserver::PerfPredictorWebContentsObserver(
+    content::WebContents* web_contents)
+    : WebContentsObserver(web_contents),
+      tab_id_(SessionTabHelper::IdForTab(web_contents)) {
+  if (!tab_id_.is_valid()) {
```

Comment:
what exactly are you trying to do here? I don't think tab id is really what you want